dss_datamover/socket_communication.py

```diff
@@ -138,67 +139,68 @@ def recv_json(self, format="JSON", timeout=RECV_TIMEOUT):
         :timeout: default 60 seconds
         :return: Return received data in json format.
         """
-        msg_len = None
+        self.socket.settimeout(timeout)
         msg = "{}"
+        msg_len = 0
+        time_started = datetime.now()
 
         try:
+            # first receive message length from payload
             msg_len_in_bytes = b''
-            received_msg_len_size = 0
-            time_started = datetime.now()
-            # Iterate till we receive 10 bytes or timeout.
-            while received_msg_len_size < MESSAGE_LENGTH:
-                received_msg_len_in_bytes = self.socket.recv(MESSAGE_LENGTH - received_msg_len_size)
-                received_msg_len_size += len(received_msg_len_in_bytes)
-                msg_len_in_bytes += received_msg_len_in_bytes
-                time_spent_in_seconds = (datetime.now() - time_started).seconds
-                if time_spent_in_seconds >= timeout:
-                    raise socket.timeout("ClientSocket: Timeout ({} seconds) from recv function".format(
-                        time_spent_in_seconds))
-            if msg_len_in_bytes != b'':
-                msg_len = int(msg_len_in_bytes)
-        except socket.timeout as e:
-            raise e
-        except socket.error as e:
-            raise socket.error("ClientSocket: Incorrect message length - {}".format(e))
-        except ValueError as e:
-            raise ValueError("ClientSocket: ValueError - {}".format(e))
+            msg_len_in_bytes = self.socket.recv(MESSAGE_LENGTH)
+            msg_len = int(msg_len_in_bytes)
+            if len(msg_len_in_bytes) != MESSAGE_LENGTH:
```

Comment:
example 
msg_len = 131
msg_len_in_bytes = 0000000131
MESSAGE_LENGTH = 10

Reply:
I am saying 
`
if msg_len != MESSAGE_LENGTH
`
You already have msg_len variable which is saving the size of the data.

Reply:
msg_len is not the size of the bytes received, it is the numerical value of how many bytes the payload is. MESSAGE_LENGTH represents the # of bytes expected to be recevied. Hence comparing msg_len to MESSAGE_LENGTH will not work. I know it is confusing because of the naming convention unfortunately.